lib/resolveTaskFn.js

```diff
@@ -101,9 +104,29 @@ export const resolveTaskFn = ({
   debugLog('execaOptions:', execaOptions)
 
   return async (ctx = getInitialState()) => {
-    const result = await (shell
+    let isExecutionDone = false
+    const execaChildProcess = shell
       ? execa.command(isFn ? command : `${command} ${files.join(' ')}`, execaOptions)
-      : execa(cmd, isFn ? args : args.concat(files), execaOptions))
+      : execa(cmd, isFn ? args : args.concat(files), execaOptions)
+
+    const interruptExecutionOnError = async () => {
```

Comment:
Can this be moved out of the function body, maybe with added parameters? Looks like code that is called many times.

Reply:
I did originally think about extracting it. And because of your various feedback, I realized what would be a good way to go about it. Thanks!
(Hope that you will agree that it's a decent improvement. 🙂)